nucliadb/nucliadb/writer/api/v1/services.py

```diff
@@ -226,21 +232,23 @@ async def set_labels(request: Request, kbid: str, labelset: str, item: LabelSet)
 )
 @requires(NucliaDBRoles.WRITER)
 @version(1)
-async def delete_labels(request: Request, kbid: str, labelset: str):
-    ingest = get_ingest()
-    pbrequest: DelLabelsRequest = DelLabelsRequest()
-    pbrequest.kb.uuid = kbid
-    pbrequest.id = labelset
-    status: OpStatusWriter = await ingest.DelLabels(pbrequest)  # type: ignore
-    if status.status == OpStatusWriter.Status.OK:
-        return None
-    elif status.status == OpStatusWriter.Status.NOTFOUND:
+async def delete_labelset_endpoint(request: Request, kbid: str, labelset: str):
+    try:
+        await delete_labelset(kbid, labelset)
+    except KnowledgeBoxNotFound:
         raise HTTPException(status_code=404, detail="Knowledge Box does not exist")
-    elif status.status == OpStatusWriter.Status.ERROR:
+    except ConflictError:
         raise HTTPException(
-            status_code=500, detail="Error on deleting labels from a Knowledge box"
+            status_code=409,
+            detail="Conflict deleting labelset from a Knowledge box. Please retry.",
         )
```

Comment:
Same here, where this exception comes from?

Reply:
https://github.com/nuclia/nucliadb/blob/main/nucliadb/nucliadb/common/maindb/tikv.py#L147